Lab1 solutions

labeebmir11 opened the pull request.

labeebmir11 committed:
Lab1 solutions

anam152 commented on Lab1/q1selectionsort.c:
Avoid Unnecessary Swaps:
Modify the code to perform swaps only when k != i, reducing redundant operations when the smallest element is already in its correct position:
if (k != i) {
    swap(&A[k], &A[i]);
}

Improve Readability:
Use descriptive variable names (minIndex instead of k) to enhance code clarity.

Dynamic Input:
Allow the user to enter the array size and elements dynamically instead of hardcoding the array for better usability.

labeebmir11 committed:
Updated Lab1 solutions as directed

labeebmir11 committed:
Updated Lab1 solutions as directed

labeebmir11 commented:
All the changes directed in the feedback have been made.

Key Modifcations:

- Dynamic input for array size and array elements.
- Added separate Display function.
- Modified variable name for better readability.
- Selection Sort: 

    1. Replaced while loop with slightly more efficient for loop.
    2. Added conditional swap to prevent unnecessary swapping.

anam152 commented on Lab1/q2insertionsort.c:
Remove Unused Functions:
The `swap` function is unnecessary for this implementation and should be removed to streamline the code.

Instead of a fixed array size, consider dynamically allocating the array and reading its size from user input for increased flexibility.

Test the program with edge cases, such as an empty array or an array with all identical elements.

labeebmir11 replied:
- Removed swap function from insertion sort.
- Added Dynamic Memory Allocation for array based on user input.
- Tested the program with edge cases.

labeebmir11 committed:
Updated Lab1 solutions
